File: GeneratedEntityReaderClasses/Components/SetStartVelocityComponent.cs
namespace GeneratedNoitaClasses;
public class SetStartVelocityComponent : NoitaComponentBase,  iNoitaType<SetStartVelocityComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaCVector2<NoitaFloat> velocity { get; set; } = new();
	public ValueRange randomize_angle { get; set; } = new();
	public ValueRange randomize_speed { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		velocity.Read(s);
		randomize_angle.Read(s);
		randomize_speed.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		velocity.Write(s);
		randomize_angle.Write(s);
		randomize_speed.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/SetStartVelocityComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaCVector2
from BasicNoitaTypes import NoitaFloat
from GeneratedClasses.ValueRange import ValueRange

from NoitaStream import NoitaStream
@dataclass
class SetStartVelocityComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    velocity:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    randomize_angle:ValueRange = field(default_factory=ValueRange)
    randomize_speed:ValueRange = field(default_factory=ValueRange)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.velocity.Read(s)
        self.randomize_angle.Read(s)
        self.randomize_speed.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.velocity.Write(s)
        self.randomize_angle.Write(s)
        self.randomize_speed.Write(s)

